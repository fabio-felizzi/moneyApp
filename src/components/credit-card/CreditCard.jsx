import React from 'react';
import PropTypes from 'prop-types';

const CreditCard = ({
    className, name, apr, balanceTransfer, purchaseOffer, creditAvailable,
}) => (
    <div className={`card ${className}`}>
        <h3>{name}</h3>
        <ul>
            <li>{`APR: ${apr}%`}</li>
            <li>{`Balance Transfer Offer Duration: ${balanceTransfer} Months`}</li>
            <li>{`Purchase Offer Duration: ${purchaseOffer} Months`}</li>
            <li>{`Credit Available: Â£${creditAvailable}`}</li>
        </ul>
    </div>
);

CreditCard.propTypes = {
    className: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    apr: PropTypes.number.isRequired,
    balanceTransfer: PropTypes.number.isRequired,
    purchaseOffer: PropTypes.number.isRequired,
    creditAvailable: PropTypes.number.isRequired,
};

export default CreditCard;
