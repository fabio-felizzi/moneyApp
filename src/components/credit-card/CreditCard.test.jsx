import React from 'react';
import { shallow } from 'enzyme';
import CreditCard from './CreditCard';

const componentProps = {
    className: 'credit-card',
    name: 'test-card',
    apr: 25,
    balanceTransfer: 6,
    purchaseOffer: 6,
    creditAvailable: 5000,
};

const componentShallow = shallow(<CreditCard {...componentProps} />);

describe('CreditCard component', () => {
    describe('Component Shapshot', () => {
        test('should match stored snapshot', () => {
            expect(componentShallow).toMatchSnapshot();
        });
    });

    describe('Component Render', () => {
        test('should render the title', () => {
            expect(componentShallow.find('h3')).toHaveLength(1);
            expect(componentShallow.find('h3').text()).toBe(componentProps.name);
        });

        test('should render a ul', () => {
            expect(componentShallow.find('ul')).toHaveLength(1);
        });

        test('should render an li with the apr prop', () => {
            expect(componentShallow.find('#apr').text()).toBe(`APR: ${componentProps.apr}%`);
        });

        test('should render an li with the balanceTransfer prop', () => {
            expect(componentShallow.find('#balance').text()).toBe(`Balance Transfer Offer Duration: ${componentProps.balanceTransfer} Months`);
        });

        test('should render an li with the purchaseOffer prop', () => {
            expect(componentShallow.find('#purchase').text()).toBe(`Purchase Offer Duration: ${componentProps.purchaseOffer} Months`);
        });

        test('should render an li with the creditAvailable prop', () => {
            expect(componentShallow.find('#credit').text()).toBe(`Credit Available: Â£${componentProps.creditAvailable}`);
        });
    });
});
