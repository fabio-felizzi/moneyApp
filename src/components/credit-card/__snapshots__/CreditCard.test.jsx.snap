// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component Component Shapshot should match stored snapshot 1`] = `
<div
  className="card credit-card"
>
  <h3>
    test-card
  </h3>
  <ul>
    <li
      id="apr"
    >
      APR: 25%
    </li>
    <li
      id="balance"
    >
      Balance Transfer Offer Duration: 6 Months
    </li>
    <li
      id="purchase"
    >
      Purchase Offer Duration: 6 Months
    </li>
    <li
      id="credit"
    >
      Credit Available: Â£5000
    </li>
  </ul>
</div>
`;
